### PR TITLE
Extend the allowed protocol versions of strict servers

### PR DESCRIPTION
**Pull request by SmallJoker**
The current strict protocol check is not optimal. 0.4.10 clients can not connect to a strict 0.4.10-dev server. [untested]

Example: https://forum.minetest.net/viewtopic.php?p=157728#p157728
This pull allows clients to connect if their protocol version is in the range of +/- 1

I understand the meaning of "strict", but adding an additional setting for this is not required.
~~An other way would be to use `strict_protocol_version_checking` as "allowed range".~~
Added in a 2nd commit. [untested]

### CODE REVIEW

**Comment by nerzhul** [comment]
All of this will be reworked, this PR will become useless at 0.5 release.

**Comment by ShadowNinja** [comment]
@nerzhul: No, this isn't related to the network protocol itself, it just affects protocol version checks.  Also even if you're working on reworking some network things this can still be added seperately.  Your reworks might not be added at all since they break everything.

@SmallJoker: Please restore your branch because GitHub won't let me re-open this.

File: src/network/serverpackethandler.cpp [hunk 1]
```diff
@@ -150,13 +150,12 @@ void Server::handleCommand_Init(NetworkPacket* pkt)
...
 
-	if (g_settings->getBool("strict_protocol_version_checking")) {
-		if (net_proto_version != LATEST_PROTOCOL_VERSION) {
-			actionstream << "Server: A mismatched (strict) client tried to "
-					<< "connect from " << addr_s << std::endl;
-			DenyAccess(pkt->getPeerId(), SERVER_ACCESSDENIED_WRONG_VERSION);
-			return;
-		}
+	s16 strict_protocol_check_range = g_settings->getS16("strict_protocol_version_checking");
+	if (abs(net_proto_version - LATEST_PROTOCOL_VERSION) > strict_protocol_check_range) {
```
Comment (est31) [inline src/network/serverpackethandler.cpp]: How does this work for negative range?

**Comment by est31** [comment]
See comment otherwise :+1:

**Comment by SmallJoker** [comment]
Using `Settings::getBool` will not work because numbers != 0 are counted as `true`.

**Comment by est31** [comment]
ah I see

**Comment by paramat** [comment]
Bump.

**Comment by SmallJoker** [comment]
Un-Bump. The branch is dead and obviously there's no need for this feature yet.
Rest in peace. May it be reopened with a new pull later.

**Pull request closed without merging**